lib/config/migrations/migration.ts

```diff
@@ -0,0 +1,18 @@
+import type { RenovateConfig } from '../types';
+
+export abstract class Migration {
+  protected readonly originalConfig: RenovateConfig;
+
+  protected readonly migratedConfig: RenovateConfig;
+
+  constructor(originalConfig: RenovateConfig, migratedConfig: RenovateConfig) {
+    this.originalConfig = originalConfig;
+    this.migratedConfig = migratedConfig;
+  }
+
+  abstract migrate(): RenovateConfig;
+
+  protected delete(property: string): void {
```

Comment:
I think you can make this type a little stricter than `string`. What about `keyof RenovateConfig`?

Reply:
I just realised it's probably not possible, as `RenovateConfig` only contains the latest types.

Reply:
Yes, indeed. I also considered this idea